### PR TITLE
Added global Error Boundary that loads 500 page

### LINKED ISSUE #819: Add global Error Boundary
Add a catch-all error boundary around the entire frontend that will catch errors and redirect the user to the /500 page.

### PR DESCRIPTION
**Pull request by ojbravo**
Closes #819

### CODE REVIEW

File: src/ErrorFallback.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import FooterCfGovWrapper from './components/FooterCfGovWrapper';
+import { Error500 } from './pages/Error/Error500';
+
+function ErrorFallback(): React.ReactElement {
+  useEffect(() => {
+    const handleClick = (event: MouseEvent): void => {
+      const target = event.target as HTMLElement;
+      if (target.tagName === 'A' && target.getAttribute('href')) {
+        event.preventDefault();
+        window.location.href = target.getAttribute('href') ?? '/';
```
[the remark this answers is not in the document]
Comment (ojbravo) [reply src/ErrorFallback.tsx]: @shindigira Great suggestions! Here are my updates:
- Updated ```Error500.tsx``` to accept props for error messages/code and display Error Details.
- Removed hacky jsdom commands from ```ErrorFallback.tsx```  and instead changed ```<Link>``` to ```<a>``` in ```_shared.tsx```  so the "Visit platform homepage" button would actually load the homepage.
- Changed 500 heading to "Something went wrong" per @billhimmelsbach suggestion

Everything appears to be working smoothly!

**Review by billhimmelsbach — APPROVED** [review]
Looks good! Tested with existing and new errors and it works great! 

I added [one ticket](https://github.com/cfpb/sbl-frontend/issues/834) to make existing custom 500 errors use the error boundary instead.

**Review by billhimmelsbach — CHANGES_REQUESTED** [review]
Oop, have some git shenanigans to fix.

**Review by billhimmelsbach — APPROVED** [review]
Looks good!

**Pull request merged**